README.md

```diff
@@ -6,10 +6,21 @@ There are easier ways to install the compiler than compiling the source, you mig
 
 ## Build from source
 
-The Gren compiler is written in Haskell, so to build from source you need to have GHC 9.4 (Haskell compiler) and Cabal 3.8 (haskell build tool) installed on your system.
+This project uses [devbox](https://www.jetify.com/devbox) for managing dependencies required to build the project. If you don't want to use devbox,
+you can find a list of the requried dependencies and the commands for building the compiler in `devbox.json`.
 
-You can install these using [ghcup](https://www.haskell.org/ghcup/). By default, ghcup will install an older version of Haskell and Cabal, so you can install and set the required versions using `ghcup tui`.
+Since Gren 0.4 the compiler is implemented in two parts. The Gren-part of the compiler lives in `src`, and once built it acts
+as a frontend to the Haskell-part of the compiler.
 
-Compiling and installing the project should just be a matter of `cabal install`, after which you should be able to run the `gren` command from your command line.
+The end goal is for the entire compiler to be written in Gren.
 
-Read the `CONTRIBUTING.md` file for some helpful commands for working on the compiler itself.
+To build the compiler:
+
+1. Use `devbox run prepare-deps` to setup the required dependencies. The first time you run this it might take a while.
+2. Build the compiler with `devbox run build`.
+
+You can now execute the compiler with `node ./cli.js` or just `./cli.js`. The Gren-part of the compiler will download a pre-built
```

Comment:
You'll need to open a `devbox shell` first or else change these to start with `devbox run`